shared/naturalcrit/splitPane/splitPane.jsx

```diff
@@ -1,200 +1,121 @@
+const SplitPane = (props)=>{
+	const {
+		storageKey = 'naturalcrit-pane-split',
+		onDragFinish = ()=>{},
+		showDividerButtons = true
+	} = props;
+
+	const [isDragging, setIsDragging] = useState(false);
+	const [dividerPos, setDividerPos] = useState(null); // Initial divider position is set to `null`
+	const [moveSource, setMoveSource] = useState(false);
+	const [moveBrew, setMoveBrew] = useState(false);
+	const [showMoveArrows, setShowMoveArrows] = useState(true);
+	const [liveScroll, setLiveScroll] = useState(false);
+
+	const dividerRef = useRef(null);
+
+	// Set initial divider position and liveScroll only after mounting
+	useEffect(()=>{
+		const savedPos = window.localStorage.getItem(storageKey);
+		setDividerPos(savedPos ? parseInt(savedPos, 10) : window.innerWidth / 2);
+		setLiveScroll(window.localStorage.getItem('liveScroll') === 'true');
+
+		const handleResize = ()=>{
+			setDividerPos((pos)=>limitPosition(pos,0.1 * (window.innerWidth - 13), 0.9 * (window.innerWidth - 13))
+			);
 		};
-	},
-
-	getInitialState : function() {
-		return {
-			currentDividerPos : null,
-			windowWidth       : 0,
-			isDragging        : false,
-			moveSource        : false,
-			moveBrew          : false,
-			showMoveArrows    : true
-		};
-	},
-
-	pane1 : React.createRef(null),
-	pane2 : React.createRef(null),
-
-	componentDidMount : function() {
-		const dividerPos = window.localStorage.getItem(this.props.storageKey);
-		if(dividerPos){
-			this.setState({
-				currentDividerPos : this.limitPosition(dividerPos, 0.1*(window.innerWidth-13), 0.9*(window.innerWidth-13)),
-				userSetDividerPos : dividerPos,
-				windowWidth       : window.innerWidth
-			});
-		} else {
-			this.setState({
-				currentDividerPos : window.innerWidth / 2,
-				userSetDividerPos : window.innerWidth / 2
-			});
-		}
-		window.addEventListener('resize', this.handleWindowResize);
-
-		// This lives here instead of in the initial render because you cannot touch localStorage until the componant mounts.
-		const loadLiveScroll = window.localStorage.getItem('liveScroll') === 'true';
-		this.setState({ liveScroll: loadLiveScroll });
-	},
-
-	componentWillUnmount : function() {
-		window.removeEventListener('resize', this.handleWindowResize);
-	},
-
-	handleWindowResize : function() {
-		// Allow divider to increase in size to last user-set position
-		// Limit current position to between 10% and 90% of visible space
-		const newLoc = this.limitPosition(this.state.userSetDividerPos, 0.1*(window.innerWidth-13), 0.9*(window.innerWidth-13));
-
-		this.setState({
-			currentDividerPos : newLoc,
-			windowWidth       : window.innerWidth
-		});
-	},
-
-	limitPosition : function(x, min = 1, max = window.innerWidth - 13) {
-		const result = Math.round(Math.min(max, Math.max(min, x)));
-		return result;
-	},
-
-	handleUp : function(e){
+		window.addEventListener('resize', handleResize);
+		return ()=>window.removeEventListener('resize', handleResize);
+	}, [storageKey]);
```

Comment:
This should probably just be an empty `[]` so it only fires once, instead of any time `storageKey` updates.